src/hooks/useStrategies.tsx

```diff
@@ -20,14 +21,40 @@ import { getCompareFunctionBySortType } from 'components/strategies/overview/uti
 import { useGetMultipleTokenPrices } from 'libs/queries/extApi/tokenPrice';
 import { useStore } from 'store';
 import { SafeDecimal } from 'libs/safedecimal';
+import { useNavigate, useSearch } from 'libs/routing';
 
 export type StrategyFilterOutput = ReturnType<typeof useStrategyFilter>;
 
+export interface MyStrategiesSearch {
+  search?: string;
+}
+type Search = MyStrategiesSearch;
+
 export const useStrategyFilter = (
   strategies: StrategyWithFiat[],
   isPending: boolean
 ) => {
-  const [search, setSearch] = useState('');
+  const searchParams: Search = useSearch({ strict: false });
+  const navigate = useNavigate();
+  const search = searchParams.search || '';
+
+  const setSearch = useCallback(
+    (value: string) => {
+      navigate({
+        search: (currentSearch: MyStrategiesSearch) => {
+          if (value) return { ...currentSearch, search: value };
+          const searchCopy = structuredClone(currentSearch);
+          delete searchCopy.search;
+          return searchCopy;
+        },
```

Comment:
I think you can do that :
```suggestion
        search: (currentSearch: MyStrategiesSearch) => {
          return { ...currentSearch, search: value ?? '' };
        },
```

Reply:
Agree

Reply:
Used `Route.postSearchFilters` to remove the search param clearing logic from the navigate function